src/components/variant/Variant.tsx

```diff
@@ -1,37 +1,93 @@
-import { Card, Grid, Typography } from '@material-ui/core';
+import { Card, CardContent, Grid, Typography } from '@material-ui/core';
+import { UIBuild } from 'evergreen.js/lib/models';
 import * as React from 'react';
 import '../../styles.css';
 
+interface State {
+  name: string,
+  statusCount: { [id: string]: number }
+}
+
 class Props {
-  public name: string;
+  public build: UIBuild;
 }
 
-export class Variant extends React.Component<Props> {
+export class Variant extends React.Component<Props, State> {
+
   constructor(props: Props) {
     super(props);
+    const statusCount = {};
+    for (const task of this.props.build.Build.tasks) {
+      if (statusCount[task.status]) {
+        statusCount[task.status] = statusCount[task.status] + 1;
+      } else {
+        statusCount[task.status] = 1;
+      }
+    }
+    this.state = {
+      name: this.props.build.Build.display_name,
+      statusCount: statusCount,
+    };
   }
 
   public render() {
 
+    const VariantsByStatus = () => (
+      <Grid container={true} spacing={8}>
+        {this.orderByPriority().map(statusObj => (
+          <Grid item={true} xs={4} key={statusObj.status}>
+            <Card>
+              <CardContent className={statusObj.status}>
+                {statusObj.count}
+              </CardContent>
+            </Card>
+          </Grid>
+        ))}
+      </Grid>
+    );
+
     return (
       <Card className="variant-card">
-        <Grid container={true}>
-          <Grid className="success" item={true} xs={4}>
-            Green
-          </Grid>
-          <Grid className="started" item={true} xs={4}>
-            Yellow
-          </Grid>
-          <Grid className="unstarted" item={true} xs={4}>
-            Gray
-          </Grid>
-        </Grid>
+        <VariantsByStatus />
         <Typography variant="body1" className="variant-title">
-          {this.props.name}
-        </Typography>      
+          {this.state.name}
+        </Typography>
       </Card>
     );
   }
+
+  private compareByPriority(a: any, b: any) {
+    const displayPriority = {
+      "failed": 0,
+      "timed-out": 1,
+      "system-failed": 2,
+      "system-unresponsive": 3,
+      "setup-failed": 4,
+      "success": 5,
+      "started": 6,
+      "dispatched": 7,
+      "undispatched": 8,
+      "inactive": 9,
+    }
+    return displayPriority[a] > displayPriority[b] ? 1 : -1;
```

Comment:
nit: a slick way to do this is to just return displayPriority[a] - displayPriority[b]. I think that might be less clear though so feel free to resolve this if you prefer what you have now